[Bug] could not find my library in bridge_generated

### Describe the bug

I have the below (very simple!) code I'm trying to use. I have name = 'lib'.

The generator runs without error, but the generated code won't compile.

The error is:
```
cargo build --manifest-path=/Users/gisborne/Documents/01-projects/cozo/cozo_dart/Cargo.toml --features= --lib --target=aarch64-apple-ios-sim
   Compiling cozo_dart v0.1.0 (/Users/gisborne/Documents/01-projects/cozo/cozo_dart)
error[E0432]: unresolved import `crate::lib`
  --> src/bridge_generated.rs:14:12
   |
14 | use crate::lib::*;
   |            ^^^ could not find `lib` in the crate root

error[E0425]: cannot find function `create_cozo` in this scope
  --> src/bridge_generated.rs:40:37
   |
40 |             move |task_callback| Ok(create_cozo(api_engine, api_path, api_options))
   |                                     ^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
14 | use crate::create_cozo;
   |

error[E0425]: cannot find function `run_script` in this scope
  --> src/bridge_generated.rs:60:37
   |
```

The code is:
```
mod bridge_generated; /* AUTO INJECTED BY flutter_rust_bridge. This line may not be accurate, and you can change it according to your needs. */
use flutter_rust_bridge::*;
use cozo::*;
use std::collections::BTreeMap;

pub fn create_cozo(
    engine: String,
    path: String,
    options: String,
) -> RustOpaque<cozo::DbInstance> {
    let db = cozo::DbInstance::new_with_str(
        &engine,
        &path,
        &options,
    ).unwrap();

    return RustOpaque::new(db);
}

pub fn run_script(
    db: RustOpaque<cozo::DbInstance>,
    script: String,
    params: String, //BTreeMap<String, DataValue>,
) -> String {
    let s: String = serde_json::from_str(&script).unwrap();
    let p: BTreeMap<String, DataValue> = s;
    return db.unwrap().run_script_fold_err(
        &script,
        p,
    );
}
```

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
2023/04/25 13:19:30 [DEBUG] configs=[Opts { rust_input_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/lib.rs", dart_output_path: "/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart", dart_decl_output_path: Some("/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart"), c_output_path: ["/Users/gisborne/Documents/01-projects/cozo/ios/Runner/bridge_generated.h", "macos/Runner/bridge_generated.h"], rust_crate_dir: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart", rust_output_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs", class_name: "CozoDart", dart_format_line_length: 80, dart_enums_style: false, skip_add_mod_to_lib: false, llvm_path: ["/opt/homebrew/opt/llvm", "/usr/local/opt/llvm", "/usr/lib/llvm-9", "/usr/lib/llvm-10", "/usr/lib/llvm-11", "/usr/lib/llvm-12", "/usr/lib/llvm-13", "/usr/lib/llvm-14", "/usr/lib/", "/usr/lib64/", "C:/Program Files/llvm", "C:/msys64/mingw64"], llvm_compiler_opts: "", manifest_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/Cargo.toml", dart_root: Some("/Users/gisborne/Documents/01-projects/cozo"), build_runner: true, block_index: BlockIndex(0), skip_deps_check: false, wasm_enabled: false, inline_rust: false, bridge_in_method: true, extra_headers: "" }]
2023/04/25 13:19:30 [DEBUG] Trying to parse "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs"
2023/04/25 13:19:30 [WARN] Skipping unresolvable module io (tried /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/io.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/io/mod.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated/io.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated/io/mod.rs, )
2023/04/25 13:19:30 [DEBUG] parse_function function name: Ident { sym: create_cozo, span: bytes(226..237) }
2023/04/25 13:19:30 [DEBUG] parse_function function name: Ident { sym: run_script, span: bytes(488..498) }
2023/04/25 13:19:30 [INFO] Phase: Validate config(s)
2023/04/25 13:19:30 [INFO] Picked config: Opts { rust_input_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/lib.rs", dart_output_path: "/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart", dart_decl_output_path: Some("/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart"), c_output_path: ["/Users/gisborne/Documents/01-projects/cozo/ios/Runner/bridge_generated.h", "macos/Runner/bridge_generated.h"], rust_crate_dir: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart", rust_output_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs", class_name: "CozoDart", dart_format_line_length: 80, dart_enums_style: false, skip_add_mod_to_lib: false, llvm_path: ["/opt/homebrew/opt/llvm", "/usr/local/opt/llvm", "/usr/lib/llvm-9", "/usr/lib/llvm-10", "/usr/lib/llvm-11", "/usr/lib/llvm-12", "/usr/lib/llvm-13", "/usr/lib/llvm-14", "/usr/lib/", "/usr/lib64/", "C:/Program Files/llvm", "C:/msys64/mingw64"], llvm_compiler_opts: "", manifest_path: "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/Cargo.toml", dart_root: Some("/Users/gisborne/Documents/01-projects/cozo"), build_runner: true, block_index: BlockIndex(0), skip_deps_check: false, wasm_enabled: false, inline_rust: false, bridge_in_method: true, extra_headers: "" }
2023/04/25 13:19:30 [INFO] Phase: Parse source code to AST, then to IR
2023/04/25 13:19:30 [DEBUG] Trying to parse "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs"
2023/04/25 13:19:30 [WARN] Skipping unresolvable module io (tried /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/io.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/io/mod.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated/io.rs, /Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated/io/mod.rs, )
2023/04/25 13:19:30 [DEBUG] parse_function function name: Ident { sym: create_cozo, span: bytes(4343..4354) }
2023/04/25 13:19:30 [DEBUG] parse_function function name: Ident { sym: run_script, span: bytes(4605..4615) }
2023/04/25 13:19:30 [INFO] Phase: Transform IR
2023/04/25 13:19:30 [INFO] Phase: Generate Rust code
2023/04/25 13:19:30 [DEBUG] execute format_rust path=["/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs"]
2023/04/25 13:19:30 [DEBUG] execute command: bin=rustfmt args="/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs" current_dir=None cmd="rustfmt" "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs"
2023/04/25 13:19:31 [DEBUG] command="rustfmt" "/Users/gisborne/Documents/01-projects/cozo/cozo_dart/src/bridge_generated.rs" stdout= stderr=
2023/04/25 13:19:31 [INFO] Phase: Generate Dart code
2023/04/25 13:19:31 [DEBUG] Guessing toolchain the runner is run into
2023/04/25 13:19:31 [DEBUG] execute command: bin=sh args="-c \"flutter\" \"--version\"" current_dir=None cmd="sh" "-c" "\"flutter\" \"--version\""
Flutter 3.7.12 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4d9e56e694 (8 days ago) • 2023-04-17 21:47:46 -0400
Engine • revision 1a65d409c7
Tools • Dart 2.19.6 • DevTools 2.20.1
2023/04/25 13:19:31 [DEBUG] command="sh" "-c" "\"flutter\" \"--version\"" stdout= stderr=
2023/04/25 13:19:31 [DEBUG] Checking presence of ffi in dependencies at /Users/gisborne/Documents/01-projects/cozo
2023/04/25 13:19:31 [DEBUG] Checking presence of ffi in dependencies at /Users/gisborne/Documents/01-projects/cozo
2023/04/25 13:19:31 [DEBUG] Checking presence of ffigen in dev_dependencies at /Users/gisborne/Documents/01-projects/cozo
2023/04/25 13:19:31 [DEBUG] Checking presence of ffigen in dev_dependencies at /Users/gisborne/Documents/01-projects/cozo
2023/04/25 13:19:31 [INFO] Phase: Generating Dart bindings for Rust
2023/04/25 13:19:31 [DEBUG] execute cbindgen rust_crate_dir=/Users/gisborne/Documents/01-projects/cozo/cozo_dart c_output_path=/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmp6Jsmbg.h
2023/04/25 13:19:31 [DEBUG] cbindgen config: Config {
    header: None,
    includes: [],
    sys_includes: [
        "stdbool.h",
        "stdint.h",
        "stdlib.h",
    ],
    after_includes: Some(
        "typedef struct _Dart_Handle* Dart_Handle;",
    ),
    trailer: None,
    include_guard: None,
    pragma_once: false,
    no_includes: true,
    autogen_warning: None,
    include_version: false,
    namespace: None,
    namespaces: None,
    using_namespaces: None,
    braces: SameLine,
    line_length: 100,
    tab_width: 2,
    line_endings: LF,
    language: C,
    cpp_compat: false,
    style: Both,
    sort_by: None,
    usize_is_size_t: false,
    parse: ParseConfig {
        parse_deps: false,
        include: None,
        exclude: [],
        expand: ParseExpandConfig {
            crates: [],
            all_features: false,
            default_features: true,
            features: None,
            profile: Debug,
        },
        clean: false,
        extra_bindings: [],
    },
    export: ExportConfig {
        include: [],
        exclude: [],
        rename: {},
        pre_body: {},
        body: {},
        prefix: None,
        item_types: [],
        renaming_overrides_prefixing: false,
        mangle: MangleConfig {
            rename_types: None,
            remove_underscores: false,
        },
    },
    macro_expansion: MacroExpansionConfig {
        bitflags: false,
    },
    layout: LayoutConfig {
        packed: None,
        aligned_n: None,
    },
    function: FunctionConfig {
        prefix: None,
        postfix: None,
        must_use: None,
        args: Auto,
        rename_args: None,
        swift_name_macro: None,
        sort_by: None,
        no_return: None,
    },
    structure: StructConfig {
        rename_fields: None,
        derive_constructor: false,
        derive_eq: false,
        derive_neq: false,
        derive_lt: false,
        derive_lte: false,
        derive_gt: false,
        derive_gte: false,
        derive_ostream: false,
        associated_constants_in_body: false,
        must_use: None,
    },
    enumeration: EnumConfig {
        rename_variants: None,
        rename_variant_name_fields: SnakeCase,
        add_sentinel: false,
        prefix_with_name: false,
        derive_helper_methods: false,
        derive_const_casts: false,
        derive_mut_casts: false,
        cast_assert_name: None,
        must_use: None,
        derive_tagged_enum_destructor: false,
        derive_tagged_enum_copy_constructor: false,
        derive_tagged_enum_copy_assignment: false,
        derive_ostream: false,
        enum_class: true,
        private_default_tagged_enum_constructor: false,
    },
    constant: ConstantConfig {
        allow_static_const: true,
        allow_constexpr: true,
        sort_by: None,
    },
    defines: {},
    documentation: true,
    documentation_style: Auto,
    documentation_length: Full,
    pointer: PtrConfig {
        non_null_attribute: None,
    },
    only_target_dependencies: false,
    cython: CythonConfig {
        header: None,
        cimports: {},
    },
}
2023/04/25 13:19:31 [DEBUG] Parsing crate cozo_dart
2023/04/25 13:19:31 [INFO] Take cozo_dart::DartPort.
2023/04/25 13:19:31 [INFO] Take cozo_dart::DartPostCObjectFnType.
2023/04/25 13:19:31 [INFO] Take cozo_dart::store_dart_post_cobject.
2023/04/25 13:19:31 [INFO] Take cozo_dart::get_dart_object.
2023/04/25 13:19:31 [INFO] Take cozo_dart::drop_dart_object.
2023/04/25 13:19:31 [INFO] Take cozo_dart::new_dart_opaque.
2023/04/25 13:19:31 [INFO] Take cozo_dart::init_frb_dart_api_dl.
2023/04/25 13:19:31 [INFO] Take cozo_dart::DartCObject - opaque (Struct is not marked #[repr(C)] or #[repr(transparent)].).
2023/04/25 13:19:31 [INFO] Take cozo_dart::WireSyncReturn.
2023/04/25 13:19:31 [INFO] Take cozo_dart::wire_create_cozo.
2023/04/25 13:19:31 [INFO] Take cozo_dart::wire_run_script.
2023/04/25 13:19:31 [INFO] Take cozo_dart::new_CozoDbInstance.
2023/04/25 13:19:31 [INFO] Take cozo_dart::new_uint_8_list_0.
2023/04/25 13:19:31 [INFO] Take cozo_dart::drop_opaque_CozoDbInstance.
2023/04/25 13:19:31 [INFO] Take cozo_dart::share_opaque_CozoDbInstance.
2023/04/25 13:19:31 [INFO] Take cozo_dart::wire_CozoDbInstance.
2023/04/25 13:19:31 [INFO] Take cozo_dart::wire_uint_8_list.
2023/04/25 13:19:31 [INFO] Take cozo_dart::free_WireSyncReturn.
2023/04/25 13:19:31 [WARN] Can't find Dart_Handle. This usually means that this type was incompatible or not found.
2023/04/25 13:19:31 [WARN] Can't find Dart_Handle. This usually means that this type was incompatible or not found.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [WARN] Missing `[defines]` entry for `target_family = "wasm"` in cbindgen config.
2023/04/25 13:19:31 [DEBUG] execute ffigen c_path=/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmp6Jsmbg.h dart_path=/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpqX3irU llvm_path=["/opt/homebrew/opt/llvm", "/usr/local/opt/llvm", "/usr/lib/llvm-9", "/usr/lib/llvm-10", "/usr/lib/llvm-11", "/usr/lib/llvm-12", "/usr/lib/llvm-13", "/usr/lib/llvm-14", "/usr/lib/", "/usr/lib64/", "C:/Program Files/llvm", "C:/msys64/mingw64"]
2023/04/25 13:19:31 [DEBUG] ffigen config: 
        output: '/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpqX3irU'
        name: 'CozoDartWire'
        description: 'generated by flutter_rust_bridge'
        headers:
          entry-points:
            - '/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmp6Jsmbg.h'
          include-directives:
            - '/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmp6Jsmbg.h'
        comments: false
        preamble: |
          // ignore_for_file: camel_case_types, non_constant_identifier_names, avoid_positional_boolean_parameters, annotate_overrides, constant_identifier_names
        
        llvm-path:
           - '/opt/homebrew/opt/llvm'
           - '/usr/local/opt/llvm'
           - '/usr/lib/llvm-9'
           - '/usr/lib/llvm-10'
           - '/usr/lib/llvm-11'
           - '/usr/lib/llvm-12'
           - '/usr/lib/llvm-13'
           - '/usr/lib/llvm-14'
           - '/usr/lib/'
           - '/usr/lib64/'
           - 'C:/Program Files/llvm'
           - 'C:/msys64/mingw64'

2023/04/25 13:19:31 [DEBUG] ffigen config_file: NamedTempFile("/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpTaovZB")
2023/04/25 13:19:31 [DEBUG] Guessing toolchain the runner is run into
2023/04/25 13:19:31 [DEBUG] execute command: bin=sh args="-c \"flutter\" \"pub\" \"run\" \"ffigen\" \"--config\" \"/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpTaovZB\"" current_dir=Some("/Users/gisborne/Documents/01-projects/cozo") cmd=cd "/Users/gisborne/Documents/01-projects/cozo" && "sh" "-c" "\"flutter\" \"pub\" \"run\" \"ffigen\" \"--config\" \"/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpTaovZB\""
Running in Directory: '/Users/gisborne/Documents/01-projects/cozo'
Input Headers: [/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmp6Jsmbg.h]
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: DartCObject, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct DartCObject, usr: c:@S@DartCObject
[WARNING]: No definition found for declaration - (Cursor) spelling: DartCObject, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct DartCObject, usr: c:@S@DartCObject
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: _Dart_Handle, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct _Dart_Handle, usr: c:@S@_Dart_Handle
[WARNING]: No definition found for declaration - (Cursor) spelling: DartCObject, kind: 2, kindSpelling: StructDecl, type: 105, typeSpelling: struct DartCObject, usr: c:@S@DartCObject
[WARNING]: Generated declaration '_Dart_Handle' start's with '_' and therefore will be private.
Finished, Bindings generated in /var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpqX3irU
2023/04/25 13:19:32 [DEBUG] command=cd "/Users/gisborne/Documents/01-projects/cozo" && "sh" "-c" "\"flutter\" \"pub\" \"run\" \"ffigen\" \"--config\" \"/var/folders/05/h5p8yj9d2dd2f8nfp6dd67xm0000gn/T/.tmpTaovZB\"" stdout= stderr=
the path is "/Users/gisborne/Documents/01-projects/cozo/ios/Runner/bridge_generated.h"
the path is "macos/Runner/bridge_generated.h"
2023/04/25 13:19:32 [DEBUG] distinct_input_types=[RustOpaque(IrTypeRustOpaque { inner_rust: "cozo :: DbInstance", inner_dart: "CozoDbInstance" }), Delegate(String), Primitive(U8), PrimitiveList(IrTypePrimitiveList { primitive: U8 })]
2023/04/25 13:19:32 [DEBUG] distinct_output_types=[RustOpaque(IrTypeRustOpaque { inner_rust: "cozo :: DbInstance", inner_dart: "CozoDbInstance" }), Delegate(String), Primitive(U8), PrimitiveList(IrTypePrimitiveList { primitive: U8 })]
2023/04/25 13:19:32 [INFO] Phase: Running build_runner
2023/04/25 13:19:32 [INFO] Phase: Formatting Dart code
2023/04/25 13:19:32 [DEBUG] execute format_dart path=["/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart", "/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart"] line_length=80
2023/04/25 13:19:32 [DEBUG] execute command: bin=sh args="-c \"dart\" \"format\" \"--line-length\" \"80\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart\"" current_dir=None cmd="sh" "-c" "\"dart\" \"format\" \"--line-length\" \"80\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart\""
Formatted /Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart
Formatted /Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart
Formatted 2 files (2 changed) in 0.14 seconds.
2023/04/25 13:19:33 [DEBUG] command="sh" "-c" "\"dart\" \"format\" \"--line-length\" \"80\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_generated.dart\" \"/Users/gisborne/Documents/01-projects/cozo/lib/bridge_definitions.dart\"" stdout= stderr=
2023/04/25 13:19:33 [INFO] Success!
2023/04/25 13:19:33 [INFO] Now go and use it :)
```


### To Reproduce

Make a simple project that tries to use the library. My main.dart looks like the below.

Make cozo_dart/src and put my lib.rs above in it.

Run the generate command:

`RUST_LOG=debug flutter_rust_bridge_codegen --rust-input cozo_dart/src/lib.rs --dart-output lib/bridge_generated.dart --dart-decl-output lib/bridge_definitions.dart -c ios/Runner/bridge_generated.h -e macos/Runner`

Try to run the project.

main.dart:
```
import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';
import 'ffi.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  MyHomePage({super.key});
  late final String? result;

  Future<String> testCozo() async {
    final dir = await getApplicationDocumentsDirectory();
    final cozo = await api.createCozo(
      engine: "sqlite",
      path: ('${dir.absolute.path}/cozo.db'),
      options: '',
    );

    const script = """
    ?[a] := a in [1, 2, 3]
    """;

    return api.runScript(script: script, db: cozo, params: '');
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Cozo Test'),
      ),
      body: Center(
        child: FutureBuilder<String>(
          future: testCozo(),
          builder: (context, snapshot) {
            if (snapshot.hasData) {
              return Text(snapshot.data!);
            } else if (snapshot.hasError) {
              return Text("${snapshot.error}");
            }
            return const CircularProgressIndicator();
          },
        ),
      ),
    );
  }
}
```

### Expected behavior

_No response_

### Generated binding code

```shell
Runner-Bridging-Header.h:


#import "GeneratedPluginRegistrant.h"
#import "bridge_generated.h"
```

bridge_generated.dart:
```
// AUTO GENERATED FILE, DO NOT EDIT.
// Generated by `flutter_rust_bridge`@ 1.75.1.
// ignore_for_file: non_constant_identifier_names, unused_element, duplicate_ignore, directives_ordering, curly_braces_in_flow_control_structures, unnecessary_lambdas, slash_for_doc_comments, prefer_const_literals_to_create_immutables, implicit_dynamic_list_literal, duplicate_import, unused_import, unnecessary_import, prefer_single_quotes, prefer_const_constructors, use_super_parameters, always_use_package_imports, annotate_overrides, invalid_use_of_protected_member, constant_identifier_names, invalid_use_of_internal_member, prefer_is_empty, unnecessary_const

import "bridge_definitions.dart";
import 'dart:convert';
import 'dart:async';
import 'package:meta/meta.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
import 'package:uuid/uuid.dart';

import 'dart:convert';
import 'dart:async';
import 'package:meta/meta.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
import 'package:uuid/uuid.dart';

import 'dart:ffi' as ffi;

class CozoDartImpl implements CozoDart {
  final CozoDartPlatform _platform;
  factory CozoDartImpl(ExternalLibrary dylib) =>
      CozoDartImpl.raw(CozoDartPlatform(dylib));

  /// Only valid on web/WASM platforms.
  factory CozoDartImpl.wasm(FutureOr<WasmModule> module) =>
      CozoDartImpl(module as ExternalLibrary);
  CozoDartImpl.raw(this._platform);
  Future<CozoDbInstance> createCozo(
      {required String engine,
      required String path,
      required String options,
      dynamic hint}) {
    var arg0 = _platform.api2wire_String(engine);
    var arg1 = _platform.api2wire_String(path);
    var arg2 = _platform.api2wire_String(options);
    return _platform.executeNormal(FlutterRustBridgeTask(
      callFfi: (port_) =>
          _platform.inner.wire_create_cozo(port_, arg0, arg1, arg2),
      parseSuccessData: _wire2api_CozoDbInstance,
      constMeta: kCreateCozoConstMeta,
      argValues: [engine, path, options],
      hint: hint,
    ));
  }

  FlutterRustBridgeTaskConstMeta get kCreateCozoConstMeta =>
      const FlutterRustBridgeTaskConstMeta(
        debugName: "create_cozo",
        argNames: ["engine", "path", "options"],
      );

  Future<String> runScript(
      {required CozoDbInstance db,
      required String script,
      required String params,
      dynamic hint}) {
    var arg0 = _platform.api2wire_CozoDbInstance(db);
    var arg1 = _platform.api2wire_String(script);
    var arg2 = _platform.api2wire_String(params);
    return _platform.executeNormal(FlutterRustBridgeTask(
      callFfi: (port_) =>
          _platform.inner.wire_run_script(port_, arg0, arg1, arg2),
      parseSuccessData: _wire2api_String,
      constMeta: kRunScriptConstMeta,
      argValues: [db, script, params],
      hint: hint,
    ));
  }

  FlutterRustBridgeTaskConstMeta get kRunScriptConstMeta =>
      const FlutterRustBridgeTaskConstMeta(
        debugName: "run_script",
        argNames: ["db", "script", "params"],
      );

  DropFnType get dropOpaqueCozoDbInstance =>
      _platform.inner.drop_opaque_CozoDbInstance;
  ShareFnType get shareOpaqueCozoDbInstance =>
      _platform.inner.share_opaque_CozoDbInstance;
  OpaqueTypeFinalizer get CozoDbInstanceFinalizer =>
      _platform.CozoDbInstanceFinalizer;

  void dispose() {
    _platform.dispose();
  }
// Section: wire2api

  CozoDbInstance _wire2api_CozoDbInstance(dynamic raw) {
    return CozoDbInstance.fromRaw(raw[0], raw[1], this);
  }

  String _wire2api_String(dynamic raw) {
    return raw as String;
  }

  int _wire2api_u8(dynamic raw) {
    return raw as int;
  }

  Uint8List _wire2api_uint_8_list(dynamic raw) {
    return raw as Uint8List;
  }
}

// Section: api2wire

@protected
int api2wire_u8(int raw) {
  return raw;
}

// Section: finalizer

class CozoDartPlatform extends FlutterRustBridgeBase<CozoDartWire> {
  CozoDartPlatform(ffi.DynamicLibrary dylib) : super(CozoDartWire(dylib));

// Section: api2wire

  @protected
  wire_CozoDbInstance api2wire_CozoDbInstance(CozoDbInstance raw) {
    final ptr = inner.new_CozoDbInstance();
    _api_fill_to_wire_CozoDbInstance(raw, ptr);
    return ptr;
  }

  @protected
  ffi.Pointer<wire_uint_8_list> api2wire_String(String raw) {
    return api2wire_uint_8_list(utf8.encoder.convert(raw));
  }

  @protected
  ffi.Pointer<wire_uint_8_list> api2wire_uint_8_list(Uint8List raw) {
    final ans = inner.new_uint_8_list_0(raw.length);
    ans.ref.ptr.asTypedList(raw.length).setAll(0, raw);
    return ans;
  }
// Section: finalizer

  late final OpaqueTypeFinalizer _CozoDbInstanceFinalizer =
      OpaqueTypeFinalizer(inner._drop_opaque_CozoDbInstancePtr);
  OpaqueTypeFinalizer get CozoDbInstanceFinalizer => _CozoDbInstanceFinalizer;
// Section: api_fill_to_wire

  void _api_fill_to_wire_CozoDbInstance(
      CozoDbInstance apiObj, wire_CozoDbInstance wireObj) {
    wireObj.ptr = apiObj.shareOrMove();
  }
}

// ignore_for_file: camel_case_types, non_constant_identifier_names, avoid_positional_boolean_parameters, annotate_overrides, constant_identifier_names

// AUTO GENERATED FILE, DO NOT EDIT.
//
// Generated by `package:ffigen`.
// ignore_for_file: type=lint

/// generated by flutter_rust_bridge
class CozoDartWire implements FlutterRustBridgeWireBase {
  @internal
  late final dartApi = DartApiDl(init_frb_dart_api_dl);

  /// Holds the symbol lookup function.
  final ffi.Pointer<T> Function<T extends ffi.NativeType>(String symbolName)
      _lookup;

  /// The symbols are looked up in [dynamicLibrary].
  CozoDartWire(ffi.DynamicLibrary dynamicLibrary)
      : _lookup = dynamicLibrary.lookup;

  /// The symbols are looked up with [lookup].
  CozoDartWire.fromLookup(
      ffi.Pointer<T> Function<T extends ffi.NativeType>(String symbolName)
          lookup)
      : _lookup = lookup;

  void store_dart_post_cobject(
    DartPostCObjectFnType ptr,
  ) {
    return _store_dart_post_cobject(
      ptr,
    );
  }

  late final _store_dart_post_cobjectPtr =
      _lookup<ffi.NativeFunction<ffi.Void Function(DartPostCObjectFnType)>>(
          'store_dart_post_cobject');
  late final _store_dart_post_cobject = _store_dart_post_cobjectPtr
      .asFunction<void Function(DartPostCObjectFnType)>();

  Object get_dart_object(
    int ptr,
  ) {
    return _get_dart_object(
      ptr,
    );
  }

  late final _get_dart_objectPtr =
      _lookup<ffi.NativeFunction<ffi.Handle Function(ffi.UintPtr)>>(
          'get_dart_object');
  late final _get_dart_object =
      _get_dart_objectPtr.asFunction<Object Function(int)>();

  void drop_dart_object(
    int ptr,
  ) {
    return _drop_dart_object(
      ptr,
    );
  }

  late final _drop_dart_objectPtr =
      _lookup<ffi.NativeFunction<ffi.Void Function(ffi.UintPtr)>>(
          'drop_dart_object');
  late final _drop_dart_object =
      _drop_dart_objectPtr.asFunction<void Function(int)>();

  int new_dart_opaque(
    Object handle,
  ) {
    return _new_dart_opaque(
      handle,
    );
  }

  late final _new_dart_opaquePtr =
      _lookup<ffi.NativeFunction<ffi.UintPtr Function(ffi.Handle)>>(
          'new_dart_opaque');
  late final _new_dart_opaque =
      _new_dart_opaquePtr.asFunction<int Function(Object)>();

  int init_frb_dart_api_dl(
    ffi.Pointer<ffi.Void> obj,
  ) {
    return _init_frb_dart_api_dl(
      obj,
    );
  }

  late final _init_frb_dart_api_dlPtr =
      _lookup<ffi.NativeFunction<ffi.IntPtr Function(ffi.Pointer<ffi.Void>)>>(
          'init_frb_dart_api_dl');
  late final _init_frb_dart_api_dl = _init_frb_dart_api_dlPtr
      .asFunction<int Function(ffi.Pointer<ffi.Void>)>();

  void wire_create_cozo(
    int port_,
    ffi.Pointer<wire_uint_8_list> engine,
    ffi.Pointer<wire_uint_8_list> path,
    ffi.Pointer<wire_uint_8_list> options,
  ) {
    return _wire_create_cozo(
      port_,
      engine,
      path,
      options,
    );
  }

  late final _wire_create_cozoPtr = _lookup<
      ffi.NativeFunction<
          ffi.Void Function(
              ffi.Int64,
              ffi.Pointer<wire_uint_8_list>,
              ffi.Pointer<wire_uint_8_list>,
              ffi.Pointer<wire_uint_8_list>)>>('wire_create_cozo');
  late final _wire_create_cozo = _wire_create_cozoPtr.asFunction<
      void Function(int, ffi.Pointer<wire_uint_8_list>,
          ffi.Pointer<wire_uint_8_list>, ffi.Pointer<wire_uint_8_list>)>();

  void wire_run_script(
    int port_,
    wire_CozoDbInstance db,
    ffi.Pointer<wire_uint_8_list> script,
    ffi.Pointer<wire_uint_8_list> params,
  ) {
    return _wire_run_script(
      port_,
      db,
      script,
      params,
    );
  }

  late final _wire_run_scriptPtr = _lookup<
      ffi.NativeFunction<
          ffi.Void Function(
              ffi.Int64,
              wire_CozoDbInstance,
              ffi.Pointer<wire_uint_8_list>,
              ffi.Pointer<wire_uint_8_list>)>>('wire_run_script');
  late final _wire_run_script = _wire_run_scriptPtr.asFunction<
      void Function(int, wire_CozoDbInstance, ffi.Pointer<wire_uint_8_list>,
          ffi.Pointer<wire_uint_8_list>)>();

  wire_CozoDbInstance new_CozoDbInstance() {
    return _new_CozoDbInstance();
  }

  late final _new_CozoDbInstancePtr =
      _lookup<ffi.NativeFunction<wire_CozoDbInstance Function()>>(
          'new_CozoDbInstance');
  late final _new_CozoDbInstance =
      _new_CozoDbInstancePtr.asFunction<wire_CozoDbInstance Function()>();

  ffi.Pointer<wire_uint_8_list> new_uint_8_list_0(
    int len,
  ) {
    return _new_uint_8_list_0(
      len,
    );
  }

  late final _new_uint_8_list_0Ptr = _lookup<
      ffi.NativeFunction<
          ffi.Pointer<wire_uint_8_list> Function(
              ffi.Int32)>>('new_uint_8_list_0');
  late final _new_uint_8_list_0 = _new_uint_8_list_0Ptr
      .asFunction<ffi.Pointer<wire_uint_8_list> Function(int)>();

  void drop_opaque_CozoDbInstance(
    ffi.Pointer<ffi.Void> ptr,
  ) {
    return _drop_opaque_CozoDbInstance(
      ptr,
    );
  }

  late final _drop_opaque_CozoDbInstancePtr =
      _lookup<ffi.NativeFunction<ffi.Void Function(ffi.Pointer<ffi.Void>)>>(
          'drop_opaque_CozoDbInstance');
  late final _drop_opaque_CozoDbInstance = _drop_opaque_CozoDbInstancePtr
      .asFunction<void Function(ffi.Pointer<ffi.Void>)>();

  ffi.Pointer<ffi.Void> share_opaque_CozoDbInstance(
    ffi.Pointer<ffi.Void> ptr,
  ) {
    return _share_opaque_CozoDbInstance(
      ptr,
    );
  }

  late final _share_opaque_CozoDbInstancePtr = _lookup<
      ffi.NativeFunction<
          ffi.Pointer<ffi.Void> Function(
              ffi.Pointer<ffi.Void>)>>('share_opaque_CozoDbInstance');
  late final _share_opaque_CozoDbInstance = _share_opaque_CozoDbInstancePtr
      .asFunction<ffi.Pointer<ffi.Void> Function(ffi.Pointer<ffi.Void>)>();

  void free_WireSyncReturn(
    WireSyncReturn ptr,
  ) {
    return _free_WireSyncReturn(
      ptr,
    );
  }

  late final _free_WireSyncReturnPtr =
      _lookup<ffi.NativeFunction<ffi.Void Function(WireSyncReturn)>>(
          'free_WireSyncReturn');
  late final _free_WireSyncReturn =
      _free_WireSyncReturnPtr.asFunction<void Function(WireSyncReturn)>();
}

class _Dart_Handle extends ffi.Opaque {}

class wire_uint_8_list extends ffi.Struct {
  external ffi.Pointer<ffi.Uint8> ptr;

  @ffi.Int32()
  external int len;
}

class wire_CozoDbInstance extends ffi.Struct {
  external ffi.Pointer<ffi.Void> ptr;
}

typedef DartPostCObjectFnType = ffi.Pointer<
    ffi.NativeFunction<
        ffi.Bool Function(DartPort port_id, ffi.Pointer<ffi.Void> message)>>;
typedef DartPort = ffi.Int64;
```
```


### OS

MacOS 13.2.1 on ARM

### Version of `flutter_rust_bridge_codegen`

1.75.1

### Flutter info

```shell
% flutter doctor -v
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.2.1 22D68 darwin-arm64, locale en-AU)
    • Flutter version 3.7.12 on channel stable at /Users/gisborne/dev/bin/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (8 days ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/gisborne/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • ANDROID_HOME = /Users/gisborne/Library/Android/sdk
    • Java binary at: /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1)
    • IntelliJ at /Users/gisborne/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 73.0.4
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.3)
    • IntelliJ at /Users/gisborne/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8836.41/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1)
    • IntelliJ at /Users/gisborne/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/231.8109.175/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.2)
    • IntelliJ at /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/222.4167.29/IntelliJ IDEA.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.3345.108

[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
    • IntelliJ at /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/222.4345.14/IntelliJ IDEA.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.3345.108

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8214.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
    • IntelliJ at /Volumes/Backup/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8617.56/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.76.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.27.0

[✓] Connected device (3 available)
    • iPhone 14 Pro (mobile) • C160B402-7919-474C-B99E-44A5C954B800 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 112.0.5615.137

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```


### Version of `clang++`

clang-1403.0.22.14.1

### Version of `ffigen`

7.2.10

### Additional context

I'm just starting to learn Rust, so forgive any rookie errors.

As you can see, what I'm trying to do is very simple.

Also, what I'm trying to bring into Dart is completely amazing, even revolutionary. SQL is awful. There has long been a much better alternative that has been defined and studied for a long time: Datalog. Datalog vs SQL is like Rust vs Cobol.

Cozo is the first ever proper open source full-blown Datalog product. Embeddable or as a server, and not only Datalog but graph and AI features. cozodb.org

So if anyone there cares to help me bring this to Dart, we can change the world.

It's not even difficult, I think. Its full API is not a huge amount more than the basics I'm trying to implement here.

--- a/frb_codegen/src/config/opts_parser.rs
+++ b/frb_codegen/src/config/opts_parser.rs
@@ -22,6 +22,12 @@ pub fn config_parse(mut raw: RawOpts) -> Vec<Opts> {
     use clap::error::ErrorKind;
     // rust input path(s)
     let rust_input_paths = get_valid_canon_paths(&raw.rust_input);
+    for rust_input in rust_input_paths.iter() {
+        if rust_input.contains("lib.rs") {
+            log::warn!("Do not use `lib.rs` as a Rust input. Please put code to be generated in an `api.rs` or similar.");
+            break;
+        }
+    }
 
     // dart output path(s)
     let dart_output_paths = get_valid_canon_paths(&raw.dart_output);

